staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -334,13 +334,23 @@ Loop:
 func (p *Parser) parseFilter(cur *ListNode) error {
 	p.pos += len("[?(")
 	p.consumeText()
+	quotes := 0
 Loop:
 	for {
 		switch p.next() {
 		case eof, '\n':
 			return fmt.Errorf("unterminated filter")
+		case '"':
```

Comment:
If I do this it looks like the parser doesn't know how to handle the single quote. 
```
kubectl get nodes -o jsonpath="{.items[?(@.status.nodeInfo.osImage == '')].status.nodeInfo.osImage}"

error: error parsing jsonpath {.items[?(@.status.nodeInfo.osImage == '')].status.nodeInfo.osImage}, unrecognized character in action: U+0027 '''
```